### ISSUE TITLE
Suggestion: consider using Lombok

### ISSUE THREAD

**Issue by orange-buffalo** [raphw/byte-buddy#245]
Hi,

I would like to recommend to use [Lombok](https://projectlombok.org/features/index.html) instead of manually/IDE generated code. For instance, this
```
@Override
public boolean equals(Object object) {
    if (this == object) return true;
    if (object == null || getClass() != object.getClass()) return false;
    ForSerializedValue that = (ForSerializedValue) object;
    return typeDescription.equals(that.typeDescription) && value.equals(that.value);
}

@Override
public int hashCode() {
    int result = typeDescription.hashCode();
    result = 31 * result + value.hashCode();
    return result;
}

@Override
public String toString() {
    return "Advice.DynamicValue.ForSerializedValue{" +
            "typeDescription=" + typeDescription +
            ", value='" + value + '\'' +
            '}';
}
```
would become
```
@EqualsAndHashCode
@ToString
```

The motivation is to remove boilerplate and improve readability of valuable code.

If you have already looked at this and decided not to use it, could you please share your thoughts why? 

Thank you,
Bogdan

**Comment by raphw** [comment]
I have been considering this for a while but never found the time but I will try to incorporate this into the next milestone release. Creating the methods has never been much more but the click of a button but it bloats the code. I also want to remove the `toString` methods alltogether as they account for quite a bit of the project's jar size.

**Comment by orange-buffalo** [comment]
Great, glad to hear that this is considered. If you need help with migration to Lombok, I can prepare a pull request for this.

**Comment by raphw** [comment]
Fixed on master. Thanks for the very clean PR and doing this very repetitive work. This makes the code base much cleaner, I really appreciate the effort!

**Comment by orange-buffalo** [comment]
Super! Thanks for the quick merge and feedback!

**Comment by wltjr** [comment]
Sorry @orange-buffalo I just saw this after my request for removal in #397 which it is now removed... I did not realize you had recently requested it be added. I hope it is acceptable it is now gone. Lombok is a very difficult package to package, thus my gripes.

**Comment by raphw** [comment]
The approach was right and I am glad that the manual hashCode/equals are removed. I mainliy pushed this to being able to build on Java 10/11.

**Comment by orange-buffalo** [comment]
@wltjr, it is very strange that one needs to package lombok anywhere, as it is an annotation processor library and should be used only during compilation of code it is processing. But anyway, as lombok has issues with the latest versions of java, then for sure it makes sense to remove it from the project. I am glad that the byte-buddy code is kept clean with the custom plugin that generates necessary boilerplate.

**Comment by wltjr** [comment]
@orange-buffalo I know about it being only needed at build/compile. On Gentoo things are built from source, even Java, as silly as it may sound. I am doing abnormal things. I tend to have to explain such often, no worries. I also ran into the lombok issues with latest Java. I am glad you on same page there. Along with keeping byte-buddy code clean. I just noticed this after lombok removal and just wanted to make sure it was ok. Thanks!
